### PR TITLE
[DCP] Adds utility for converting dcp to torch save format

### PR DESCRIPTION
**Pull request by LucasLLC**
Should address the `dcp -> torch.save` portion of [118207](https://github.com/pytorch/pytorch/issues/118207)

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #119335
* #119333
* __->__ #118894
* #119332
Differential Revision: [D53285982](https://our.internmc.facebook.com/intern/diff/D53285982/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

### CODE REVIEW

File: torch/distributed/checkpoint/format_utils.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            else:
+                state_dict[k] = v
+
+        super().set_up_planner(state_dict, metadata, is_coordinator)
+
+
+
+def dcp_to_torch_save(
+    dcp_checkpoint_dir: Union[str, os.PathLike],
+    torch_save_fn: Union[str, os.PathLike],
```
Comment (fegin) [inline torch/distributed/checkpoint/format_utils.py:58]: nit: some people may confused `fn` with `function`. Suggest to use other name.

File: torch/distributed/checkpoint/format_utils.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    coordinator_rank: int = 0,
+):
+    """
```
Comment (fegin) [inline torch/distributed/checkpoint/format_utils.py:61]: Do you plan to modify this API to support converting with more than 1 rank? It seems like this API always uses one rank. If that's the case, maybe we just don't assume it is SPMD-style but a single GPU API?

**Review by fegin — COMMENTED** [review]
Do you also consider to add an executable tool that use this utility function?

**Comment by LucasLLC** [comment]
Moved to https://github.com/pytorch/pytorch/pull/119816 due to CI issues

**Comment by LucasLLC** [comment]
https://github.com/pytorch/pytorch/pull/119816

**Pull request closed without merging**